app/src/main/java/com/example/tasky/agenda_details/data/repository/AgendaDetailsRemoteRepositoryImpl.kt

```diff
@@ -0,0 +1,80 @@
+package com.example.tasky.agenda_details.data.repository
+
+import android.util.Log
+import com.example.tasky.agenda_details.data.model.EventDetails
+import com.example.tasky.agenda_details.domain.model.EventResponse
+import com.example.tasky.agenda_details.domain.repository.AgendaDetailsRemoteRepository
+import com.example.tasky.common.data.remote.TaskyApi
+import com.example.tasky.common.domain.Result
+import com.example.tasky.common.domain.error.DataError
+import com.example.tasky.common.domain.util.toNetworkErrorType
+import com.example.tasky.di.AuthenticatedApi
+import com.google.gson.Gson
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.MultipartBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import okio.IOException
+
+class AgendaDetailsRemoteRepositoryImpl(
+    @AuthenticatedApi private val api: TaskyApi
+) : AgendaDetailsRemoteRepository {
+    override suspend fun createEvent(
+        eventDetails: EventDetails,
+        photosByteArray: List<ByteArray>
+    ): Result<EventResponse, DataError.Network> {
+        return try {
+            val gson = Gson()
+            val json = gson.toJson(eventDetails)
+            val requestBody = json.toRequestBody("application/json; charset=utf-8".toMediaType())
+
+            val photoParts = photosByteArray.mapIndexed { index, byteArray ->
+                val requestFile =
+                    byteArray.toRequestBody("image/jpeg".toMediaTypeOrNull(), 0, byteArray.size)
+                MultipartBody.Part.createFormData("photo$index", "image$index.jpg", requestFile)
```

Comment:
Do you let the user only pick JPEGs or could this break for PNGs?

Reply:
I'm adding a condition so Png can be allowed too